### PR TITLE
Small CI cleanup

### PR DESCRIPTION
**Pull request by alexfikl**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ Python 3 POCL:
   script: |
     export PY_EXE=python3
     export PYOPENCL_TEST=portable:pthread
-    # cython is here because pytential (for now, for TS) depends on it
-    export EXTRA_INSTALL="pybind11 cython numpy mako mpi4py oct2py"
```
Comment (alexfikl) [inline .gitlab-ci.yml:6]: These seem to have been copy pasted from [meshmode/.gitlab-ci.yml](https://github.com/inducer/meshmode/blob/main/.gitlab-ci.yml), but they're not needed here, from what I can tell.

**Comment by alexfikl** [comment]
Running the Gitlab CI at
https://gitlab.tiker.net/fikl2/arraycontext/-/pipelines/294436
since the changes are mostly there.

**Review by inducer — APPROVED** [review]
Thanks, LGTM! Ready to merge once Gitlab passes.

**Comment by inducer** [comment]
Ugh, it ran out of time. :/ Oh well, let's assume it passes. I'll mess with the time limits once merged.

**Pull request merged**